### PR TITLE
feat(!): simplify OIDC token validation errors

### PR DESCRIPTION
**Pull request by ovindu-a**
## Description

This pull request is to simplify the code for token validation by removing unnecessary error codes.

## References

Fixes issue #1827 

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by ovindu-a** [comment]
@miparnisari I have removed unnecessary error codes other than Invalid token and Invalid claim in the above commit. I also changed the tests the reflect this. Is this alright or must I further simplify the code in the `Authenticate` function. 

I will update to MicahParks/keyfunc@v3 in the next commit.

**Review by miparnisari — DISMISSED** [review]

**Comment by ovindu-a** [comment]
@miparnisari there seems to be some errors that come up when trying to move from MicahParks/keyfunc@v2 to v3. Should I address them in the same PR?

**Comment by miparnisari** [comment]
@ovindu-a let's tackle that in a separate PR :)

**Review by jpadilla — CHANGES_REQUESTED** [review]
Seems like linting check is failing. Can you try updating your branch and running `make lint`?

**Comment by ovindu-a** [comment]
@jpadilla done, is it alright now?

File: internal/authn/oidc/oidc.go [hunk 1]
```diff
@@ -40,13 +40,9 @@ type RemoteOidcAuthenticator struct {
...
 
-	errInvalidAudience = status.Error(codes.Code(openfgav1.AuthErrorCode_auth_failed_invalid_audience), "invalid audience")
-	errInvalidClaims   = status.Error(codes.Code(openfgav1.AuthErrorCode_invalid_claims), "invalid claims")
-	errInvalidIssuer   = status.Error(codes.Code(openfgav1.AuthErrorCode_auth_failed_invalid_issuer), "invalid issuer")
-	errInvalidSubject  = status.Error(codes.Code(openfgav1.AuthErrorCode_auth_failed_invalid_subject), "invalid subject")
-	errInvalidToken    = status.Error(codes.Code(openfgav1.AuthErrorCode_auth_failed_invalid_bearer_token), "invalid bearer token")
-
-	fetchJWKs = fetchJWK
+	errInvalidClaims = status.Error(codes.Code(openfgav1.AuthErrorCode_invalid_claims), "invalid claims")
+	errInvalidToken  = status.Error(codes.Code(openfgav1.AuthErrorCode_auth_failed_invalid_bearer_token), "invalid bearer token")
```
Comment (miparnisari) [inline internal/authn/oidc/oidc.go:44]: in line 96 could you return `errInvalidClaims` instead and get rid of `errInvalidToken`?

Comment (ovindu-a) [reply internal/authn/oidc/oidc.go]: I have made the above changes

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,13 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
 ### Added
 * Added `start_time` parameter to `ReadChanges` API to allow filtering by specific time [#2020](https://github.com/openfga/openfga/pull/2020)
 
+### Changed
+* Refactored the code in OIDC authenticator by removing unwanted error messages when validating a token. Token validation will now throw codes 1004 and 1005
+for invalid tokens. [#1999](https://github.com/openfga/openfga/pull/1999)
+
+### Removed
+* Removed `errInvalidToken` error type. [#1999](https://github.com/openfga/openfga/pull/1999)
+
```
Comment (miparnisari) [inline CHANGELOG.md:20]:
```suggestion
### Changed
* OIDC token validation will now exclusively throw error code 1004 for invalid tokens. [#1999](https://github.com/openfga/openfga/pull/1999)

```

**Review by miparnisari — DISMISSED** [review]

**Review by miparnisari — COMMENTED** [review]
@ovindu-a please update this failing test: https://github.com/openfga/openfga/blob/af054d9693bd7ebd0420456b144c2fb6888aaf87/cmd/run/run_test.go#L576-L582

**Review by miparnisari — APPROVED** [review]

**Review by jpadilla — APPROVED** [review]

**Pull request merged**